soundata/datasets/singapura.py

```diff
@@ -0,0 +1,379 @@
+]
+
+REMOTES: RemoteDictType = {**audio_remotes, **meta_remotes}
+
+# -- Include any information that should be printed when downloading
+# -- remove this variable if you don't need to print anything during download
+DOWNLOAD_INFO = """
+SINGA:PURA (SINGApore: Polyphonic URban Audio) v1.0a
+
+Labelled data subset downloaded from https://zenodo.org/record/5645825.
+"""
+
+# -- Include the dataset's license information
+LICENSE_INFO = "Creative Commons Attribution-ShareAlike 4.0 International"
+
+
+class Clip(core.Clip):
+    """
+    Args:
+        clip_id (str): clip id of the clip
+
+    Attributes:
+        clip_id (str): clip id
+        audio (np.ndarray, float): audio data
+        audio_path (str): path to the audio file
+        events (annotations.Events): sound events with start time, end time, label and confidence
+        annotation_path (str): path to the annotation file
+        sensor_id (str): sensor_id of the device used to record the data
+        town (str): town in Singapore where the sensor is located
+        timestamp (np.datetime): timestamp of the recording
+        dotw (int): day of the week when the clip was recorded, starting from 0 for Sunday
+    """
+
+    def __init__(self, clip_id, data_home, dataset_name, index, metadata):
+        super().__init__(
+            clip_id,
+            data_home,
+            dataset_name=dataset_name,
+            index=index,
+            metadata=metadata,
+        )
+
+        self.audio_path = self.get_path("audio")
+        self.annotation_path = self.get_path("annotation")
+
+    @core.cached_property
+    def events(self) -> Optional[annotations.Events]:
+        """
+        The clip's event annotations
+
+        Returns:
+            * annotations.Events - sound events with start time, end time, label and confidence
+        """
+        return load_annotation(self.annotation_path)
+
+    @property
+    def audio(self):
+        """
+        The clip's audio
+
+        Returns:
+            * np.ndarray - audio signal
+        """
+        return load_audio(self.audio_path)
+
+    @property
+    def sensor_id(self) -> str:
+        """
+        The clip's sensor ID
+
+        Returns:
+            * str - sensor_id of the device used to record the data
+        """
+        return self._clip_metadata["sensor_id"]
+
+    @property
+    def town(self) -> str:
+        """
+        The clip's location
+
+        Returns:
+            * str - location of the sensor, one of {'East 1', 'East 2', 'West 1', 'West 2'}
+        """
+        return self._clip_metadata["sensor_id"]
+
+    @property
+    def timestamp(self) -> np.datetime64:
+        """
+        The clip's timestamp
+
+        Returns:
+            * np.datetime64 - timestamp of the clip
+        """
+
+        if self._clip_metadata["timezone"] == "SGT":
+            tz = "+08:00"
+        else:
+            # currently all of our data are in GMT+08:00
+            raise NotImplementedError
```

Comment:
Are you adding the `if` statement thinking of adding more data in other timezones in the future? (asking because otherwise it won't be needed). If so, maybe you can add a comment explaining that data is in `SGT` when you raise the exception?

Also Line 274 is not covered by tests right now, could you add a simple test there?

Reply:
All of our data will be in SGT since we are only deploying in Singapore. I will remove the `if`.